### PR TITLE
BUGFIX: PSR-7 compatibility for backend module

### PR DESCRIPTION
**Pull request by Sebobo**
Resolves: #32

### CODE REVIEW

File: Classes/Domain/Dto/AbstractDataResult.php [hunk 1]
```diff
@@ -30,7 +31,7 @@ public function __construct(array $results)
     /**
      * @return array
      */
-    function jsonSerialize() {
+    function jsonSerialize(): array {
```
Comment (kdambekalns) [inline Classes/Domain/Dto/AbstractDataResult.php]:
```suggestion
    public function jsonSerialize(): array {
```

File: Classes/Domain/Dto/BrowserDataResult.php [hunk 2]
```diff
@@ -14,7 +16,7 @@ class BrowserDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/BrowserDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Domain/Dto/ColumnDataResult.php [hunk 3]
```diff
@@ -15,7 +17,7 @@ class ColumnDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/ColumnDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Domain/Dto/DeviceDataResult.php [hunk 4]
```diff
@@ -15,7 +17,7 @@ class DeviceDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/DeviceDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Domain/Dto/OperatingSystemDataResult.php [hunk 5]
```diff
@@ -16,7 +17,7 @@ class OperatingSystemDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/OperatingSystemDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Domain/Dto/OutlinkDataResult.php [hunk 6]
```diff
@@ -15,7 +17,7 @@ class OutlinkDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/OutlinkDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Domain/Dto/TimeSeriesDataResult.php [hunk 7]
```diff
@@ -15,7 +17,7 @@ class TimeSeriesDataResult extends AbstractDataResult
     /**
      * {@inheritdoc}
      */
-    function jsonSerialize()
+    function jsonSerialize(): array
```
Comment (kdambekalns) [inline Classes/Domain/Dto/TimeSeriesDataResult.php]:
```suggestion
    public function jsonSerialize(): array
```

File: Classes/Service/Reporting.php [hunk 8]
```diff
@@ -174,22 +185,23 @@ protected function getLiveNodeUri(NodeInterface $liveNode, ControllerContext $co
...
+     * @param Uri $apiCallUrl
      * @param bool $useCache
      * @param integer $cacheLifetime of this entry in seconds. If NULL is specified, the default lifetime is used. "0" means unlimited lifetime.
      * @return array|null the json decoded content of the api response or null if an error occurs
      */
-    protected function request($apiCallUrl, $useCache = true, $cacheLifetime = null)
+    protected function request(Uri $apiCallUrl, bool $useCache = true, ?int $cacheLifetime = null): ?array
     {
-        $cacheIdentifier = sha1($apiCallUrl);
+        $cacheIdentifier = sha1($apiCallUrl->__toString());
```
Comment (kdambekalns) [inline Classes/Service/Reporting.php]:
```suggestion
        $cacheIdentifier = sha1((string)$apiCallUrl);
```

File: Classes/Service/Reporting.php [hunk 9]
```diff
@@ -108,11 +117,12 @@ public function getNodeStatistics(NodeInterface $node = NULL, ControllerContext
             try {
                 $pageUrl = $this->getLiveNodeUri($liveNode, $controllerContext)->__toString();
             } catch (\Exception $e) {
-                $this->systemLogger->log($e->getMessage(), LOG_WARNING);
+                $this->logger->warning($e->getMessage());
```
Comment (kdambekalns) [inline Classes/Service/Reporting.php]:
```suggestion
                $this->logger->warning($e->getMessage(), \Neos\Flow\Log\Utility\LogEnvironment::fromMethodName(__METHOD__));
```

File: Classes/Service/Reporting.php [hunk 10]
```diff
@@ -174,22 +185,23 @@ protected function getLiveNodeUri(NodeInterface $liveNode, ControllerContext $co
...
         if ($useCache) {
             try {
                 $cachedResults = $this->apiCache->get($cacheIdentifier);
                 if (is_array($cachedResults)) {
                     return $cachedResults;
                 }
             } catch (\Exception $e) {
-                $this->systemLogger->log($e->getMessage(), LOG_WARNING);
+
+                $this->logger->warning($e->getMessage());
```
Comment (kdambekalns) [inline Classes/Service/Reporting.php]:
```suggestion
                $this->logger->warning($e->getMessage(), \Neos\Flow\Log\Utility\LogEnvironment::fromMethodName(__METHOD__));
```

File: Classes/Service/Reporting.php [hunk 11]
```diff
@@ -199,14 +211,14 @@ protected function request($apiCallUrl, $useCache = true, $cacheLifetime = null)
...
-                $results = json_decode($response->getContent(), true);
+                $results = json_decode($response->getBody()->getContents(), true);
                 if ($useCache) {
                     $this->apiCache->set($cacheIdentifier, $results, [], $cacheLifetime);
                 }
                 return $results;
             }
         } catch (\Exception $e) {
-            $this->systemLogger->log($e->getMessage(), LOG_WARNING);
+            $this->logger->warning($e->getMessage());
```
Comment (kdambekalns) [inline Classes/Service/Reporting.php]:
```suggestion
            $this->logger->warning($e->getMessage(), \Neos\Flow\Log\Utility\LogEnvironment::fromMethodName(__METHOD__));
```

**Comment by pascualet** [comment]
Great, thank you very much @Sebobo and @kdambekalns 
It's working fine now.

**Pull request merged**